lib/fs/basicfs_watch.go

```diff
@@ -78,7 +83,14 @@ func (f *BasicFilesystem) watchLoop(ctx context.Context, name string, roots []st
 
 		select {
 		case ev := <-backendChan:
-			relPath, err := f.unrootedChecked(ev.Path(), roots)
+			evPath := ev.Path()
+
+			if !utf8.ValidString(evPath) {
+				l.Debugln(f.Type(), f.URI(), "Watch: Ignoring invalid UTF-8")
+				return
```

Comment:
This should be a continue - it's not fatal to get an event for a file with invalid utf-8.

Reply:
Sorry about that

Reply:
No need to be sorry ;)